ci/prepare.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+# assumes build host is linux 64bit and either ubuntu >= 16.04, debian > 8
+# fedora/centos/redhat are minimally supported by allowing multilib validation
+# alpine is enabled for multiple compiler tests
+
+usage() {
+  echo "usage: $0 [--help] [windows]"
+  echo
+  echo "windows: installs wine (EXPENSIVE)"
+  exit 1
+}
+
+if [ $# -gt 0 ]; then
+  if [ $# -gt 1 ] || [ "$1" != "windows" ]; then
+    usage
+  else
+    WINE=1
+  fi
+fi
+
+if [ -e /etc/os-release ]; then
+  . /etc/os-release
+else
+  ID=unknown
+fi
+
+ARCH=$(uname -m)
+if [ $? -eq 0 ] && [ "$ARCH" != "x86_64" ]; then
+  if [ "$ID" != "alpine" ]; then
+    exit 0
+  fi
+fi
+
+USERID=$(id -u)
+if [ $? -eq 0 ] && [ $USERID -gt 0 ]; then
+  type sudo >/dev/null
+  if [ $? -eq 0 ]; then
+    SUDO=sudo
+  else
+    echo "error: need root privileges"
+    exit 1
+  fi
+fi
+
+case $ID in
+  debian|ubuntu)
+    $SUDO apt-get update
+    $SUDO apt-get install -y gcc make
+    $SUDO apt-get install -y qemu-user-static
+    $SUDO apt-get install -y gcc-aarch64-linux-gnu gcc-arm-linux-gnueabihf
+    $SUDO apt-get install -y gcc-multilib-sparc64-linux-gnu
+    if [ "$VERSION_ID" != "19.10" ]; then
+      # mips in ubuntu 19.10 is beyond repair
+      $SUDO apt-get install -y gcc-mips-linux-gnu gcc-mips64el-linux-gnuabi64
+    fi
+    $SUDO apt-get install -y gcc-powerpc-linux-gnu gcc-powerpc64le-linux-gnu
```

Comment:
the list of architectures chosen include big and little endian but is missing soft float

considering all different mips versions, there might be the need to add tests or even architectures there